numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -271,6 +271,13 @@ PyArray_AxisConverter(PyObject *obj, int *axis)
         if (error_converting(*axis)) {
             return NPY_FAIL;
         }
+        if(*axis == NPY_MAXDIMS){
+            /* NumPy 1.23, 2022-01-08 */
+            if (DEPRECATE("axis = 32 is deprecated "
+                          "please use axis = None instead. ") < 0) {
```

Comment:
I am considering to rephrase this slightly, although it seems fine, hardcoding 32 is also fine (DEPRECATIONS are a bit annoying with formatting).

Maybe:
> Using 32 as an axis is deprecated, it has the same meaning as `None` which should be used instead.  (Deprecated NumPy 1.23)

But not sure. I would like the `(Deprecated NumPy 1.23)` though.

Reply:
`np.MAXDIMS` might be better than `32`; or perhaps both.